### PR TITLE
Add support for Anthropic models via AWS Bedrock

### PR DESCRIPTION
**Pull request by ABarLT**
This adds support for Anthropic models via AWS Bedrock. It has been tested with Claude 3.5 Sonnet, but should also support 3 Opus and 3 Haiku as well. 
This is done by adding bedrock as an endpoint and implementing endpointBedrock API. Multimodality (image upload functionality) is also supported.
Co-authored-by: Zac Pullar-Strecker <zac.pullar-strecker@lanzatech.com>

### CODE REVIEW

**Comment by ABarLT** [comment]
> Looks like there's some type checking issues, could you take a look when you have the time?

Thank you for reviewing this! I fixed the type errors.

File: src/lib/server/endpoints/aws/endpointBedrock.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+						generated_text: null,
+						details: null,
+					} satisfies TextGenerationStreamOutput;
+				} else if (chunk_type === "message_stop") {
+					yield {
+						token: {
+							id: tokenId++,
+							text: "",
+							logprob: 0,
+							special: false,
```
Comment (nsarrazin) [inline src/lib/server/endpoints/aws/endpointBedrock.ts:103]: I think you can get around your issue above by setting `special: true` in the last message

File: src/lib/components/chat/ChatWindow.svelte [hunk 2]
```diff
@@ -325,7 +325,7 @@
 						{#if isFileUploadEnabled}
 							<UploadBtn bind:files mimeTypes={activeMimeTypes} classNames="ml-auto" />
 						{/if}
-						{#if messages && lastMessage && lastMessage.interrupted && !isReadOnly}
+						{#if messages && lastMessage && lastMessage.interrupted && !isReadOnly && !currentModel.id.includes("claude")}
```
Comment (nsarrazin) [inline src/lib/components/chat/ChatWindow.svelte:328]: Something is probably wrong in the endpoint code if you have to do this, see my other comment

File: src/lib/server/endpoints/aws/endpointBedrock.ts (hunk 1 above)
Comment (ABarLT) [reply src/lib/server/endpoints/aws/endpointBedrock.ts]: Yes, that fixed it. Thank you!

**Comment by nsarrazin** [comment]
Thanks for the contribution :rocket:

**Review by nsarrazin — APPROVED** [review]

**Comment by ABarLT** [comment]
> Thanks for the contribution 🚀
Thank you!! And Zac Pullar-Strecker [zac.pullar-strecker@lanzatech.com] also helped to co-author this (I think this is missing in the merged commit) :)

**Comment by nsarrazin** [comment]
Oof sorry, didn't see :disappointed: thanks folks for the PR!!

**Pull request merged**